### PR TITLE
add footprint categories to pledge dashboard and form

### PR DESCRIPTION
**Pull request by 0xAeterno**
## Description

Add footprint categories to pledge dashboard 🎉 
- Add categories field array to form
- Total footprint is calculated based on the sum of entered category totals
- Currently limited to 10 maximum categories (not sure how to programmatically color code chart sectors with nice colors consistently)
- Render user footprint categories onto `<FootprintCard />`
- Doughnut chart to visualise footprint breakdown
    - Tooltip and active states on hover
    - Charts are loaded dynamically 
- Page headers
    
Refactor
- Adapt pledge to form value requirements in `<PledgeForm />`
- Refactored `<PageHeader />` with sensible defaults

TODO
- [ ] Finalise color palette for chart (@Go-Mori)
- [x] Update firebase footprint attribute when `total` is same as previous but `categories` object is different
- [ ] Should we use klima infinity branding image for pledge page's meta tag (@Atmosfearful)
- [x] Pledge index basic input state styling

## Related Ticket
Linked to epic ticket: https://github.com/KlimaDAO/klimadao/issues/255

## Other
If bundle size is too large (due to charting library), we can potentially look into [tree shaking](https://github.com/recharts/recharts/issues/2174) the module.
It is possible to modularise the imports but we'll have to redefine the typedefs locally.
```ts
import Cell from 'recharts/es6/component/Cell';
```

## Changes

| Screen | Full  | Mobile |
|-------|---------|-----|
| Footprint card (with data) | image | image
| Footprint card (empty state) |image| image |
| Form |image|image|
| Form (field array cleared) | image| |

## Checklist

- [x] Building the site with `npm run build-site` works without errors
- [x] Building the app with `npm run build-app` works without errors
- [x] I formatted JS and TS files with running `npm run format-all`

### CODE REVIEW

**Comment by Atmosfearful** [comment]
#255

File: site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/FootprintCharts.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import { Text } from "@klimadao/lib/components";
+
+import { CategoryWithPercent } from ".";
+import * as styles from "./styles";
+
+export type FootprintChartProps = {
+  data: CategoryWithPercent[];
+};
+
+const renderActiveIndex = (props: SectorProps) => (
```
Comment (0xAeterno) [inline site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/FootprintCharts.tsx:20]: function call type inferred from <Pie />

File: site/components/pages/Pledge/styles.ts [hunk 2]
```diff
@@ -20,6 +20,7 @@ export const container = css`
 
 export const errorMessage = css`
   color: var(--warn);
+  // margin-bottom: 1rem;
```
Comment (Atmosfearful) [inline site/components/pages/Pledge/styles.ts:23]:
```suggestion
```

File: site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/FootprintSkeleton.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+import * as styles from "./styles";
+
+type SkeletonProps = {
+  width: number;
+};
+
+const Skeleton: FC<SkeletonProps> = (props) => (
+  <div
+    className={css({
```
Comment (Atmosfearful) [inline site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/FootprintSkeleton.tsx:12]: you already have `./styles`, why inline this?

**Review by Atmosfearful — APPROVED** [review]
looks good! Merging. Be sure to update this line in a follow up PR to support relative path so it works on staging. https://github.com/KlimaDAO/klimadao/blob/1a067b8cc99c62330c13fb6d7631ce2cdfc12389/site/components/pages/Pledge/lib/putPledge.ts#L11

For future reference we only need to use this `API_URL` constant for the dapp, as the dapp can't have it's own nextjs API endpoints (so it needs to reference the prod `site` url).

**Pull request merged**